Wells

waqashussain117 opened the pull request:
Fixed well import issue name.
Added right click new menu i.e., Open Mesh slicer (tested and worked).
The branch was merged from main before creating pull request and testing it two times both on well import and mesh slicer dialog.

andrea-bistacchi commented:
Hi there are still problems.

- To stick to the standard, "name" should remain "name", not "Loc ID". A the moment this creates a duplicate in the collection dataframe.

<img width="1711" height="226" alt="image" src="https://github.com/user-attachments/assets/d3eb76a6-8e18-409c-881f-fa9033426dfc" />

- The commit about mesh slicer should have gone in that branch (but, ok, this is not a big issue).

- If you change the shown property with cylinder or flag log view on, then the cylinder/flag remains there forever, also when you try to completely hide the well (checkbox in the tree) and the only solutions are (i) to kill the view, or (ii) to switch some times properties and cylinder vs flag, until something changes (but in any case the flag/cylinder cannot be hidden).

<img width="892" height="806" alt="image" src="https://github.com/user-attachments/assets/7d630dc0-3d98-4a0c-9250-b4a9565becf3" />

waqashussain117 commented:
<img width="1096" height="720" alt="image" src="https://github.com/user-attachments/assets/4f29c3c1-d45a-44dc-b605-ca5f85c99519" />
<img width="880" height="665" alt="image" src="https://github.com/user-attachments/assets/a5859e8d-2138-4f79-b682-042b4ff4036d" />
<img width="877" height="796" alt="image" src="https://github.com/user-attachments/assets/712b04c2-0bf0-4eaf-9fab-28a6565579c6" />
<img width="952" height="774" alt="image" src="https://github.com/user-attachments/assets/523dc9e5-2d1b-4870-a9cb-816a43f14910" />
<img width="1126" height="191" alt="image" src="https://github.com/user-attachments/assets/b0346745-39cf-4c38-b30c-1150ded9aa5f" />
@andrea-bistacchi All the problems are now fixed, removed the loc id from the collection and the code base.
Fixed the visibility issue by adding tree toggles for property to hide the wells properly

andrea-bistacchi commented:
> I partially fixed the problems above with [this commit](https://github.com/gecos-lab/PZero/commit/423dacc2cc018511a429d5259788c6cecc5571d0) Now it is possible to show `LITHOLOGY` as a property, but it's not possible to change its colours.
> <img alt="image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/150252648/519190047-97ebadfb-09a5-4ef2-873e-7b0f6f496e4e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQxNzE2MTgsIm5iZiI6MTc2NDE3MTMxOCwicGF0aCI6Ii8xNTAyNTI2NDgvNTE5MTkwMDQ3LTk3ZWJhZGZiLTA5YTUtNGVmMi04NzNlLTdiMGY2ZjQ5NmU0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyNlQxNTM1MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDRlMzg5YjBiZGQ4M2NmM2FjNmU2MjE3MWU5NzU1MzE4ZmIzNWZhMDgxYmRkZTk1ZWZlODY2Njg4ZGQyMDkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BuSFpie32Jovkb9v3ofi1llYDLDhRnUaCWy9Icai0ng">



What does it mean exactly? I guess that the problem here is that we need to implement categorical properties...

andrea-bistacchi reviewed (changes requested):
we must fix all problemas before merging

r-monti14 commented:
> > I partially fixed the problems above with [this commit](https://github.com/gecos-lab/PZero/commit/423dacc2cc018511a429d5259788c6cecc5571d0) Now it is possible to show `LITHOLOGY` as a property, but it's not possible to change its colours.
> > <img alt="image" width="1920" height="1080" src="https://private-user-images.githubusercontent.com/150252648/519190047-97ebadfb-09a5-4ef2-873e-7b0f6f496e4e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjQxNzE2MTgsIm5iZiI6MTc2NDE3MTMxOCwicGF0aCI6Ii8xNTAyNTI2NDgvNTE5MTkwMDQ3LTk3ZWJhZGZiLTA5YTUtNGVmMi04NzNlLTdiMGY2ZjQ5NmU0ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTEyNlQxNTM1MThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDRlMzg5YjBiZGQ4M2NmM2FjNmU2MjE3MWU5NzU1MzE4ZmIzNWZhMDgxYmRkZTk1ZWZlODY2Njg4ZGQyMDkyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BuSFpie32Jovkb9v3ofi1llYDLDhRnUaCWy9Icai0ng">
> 
> What does it mean exactly? I guess that the problem here is that we need to implement categorical properties...

As you can see in the picture, `LITHOLOGY` is not in `Properties`

andrea-bistacchi commented:
OK.

1) have you fixed the "name" vs. "loc_id"? Not clear from the PR report...

2) I am merging the [parent_uid branch](https://github.com/gecos-lab/PZero/tree/parent_uid), and then I'm working on cross sections refactoring #209 right now, so please update all before placing a new PR regarding this branch on wells.

waqashussain117 commented:
@andrea-bistacchi Name vs loc id was fixed
I stopped `well2vtk` from inserting a `Loc ID` key when registering wells; `name` is now the single identifier that enters the collection.  Also any wells with loc id is mapped back to name only in the case if name is missing and the extra key is dropped. It stopped the creation of duplicate columns as discussed above in the screen shot
    bore_obj_attributes = deepcopy(WellCollection().entity_dict)
    bore_obj_attributes["uid"] = well_uid
    bore_obj_attributes["name"] = well_obj.ID
    bore_obj_attributes["topology"] = "PolyLine"
    ...
    self.well_coll.add_entity_from_dict(entity_dict=bore_obj_attributes)
    
also updated legend manager to look for name so ui doesnt depend on non standard field

        updated_list = parent.well_coll.df.loc[
            parent.well_coll.df["name"] == locid, "uid"
        ].to_list()
        
        
 Screenshot is attached above
